scripts/stemcell-repack-debuger.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+set -e
+
+show_help() {
+    echo "Usage: $0 [options]"
+    echo
+    echo "Options:"
+    echo "  --help, -h                Show this help message and exit"
+    echo "  --debug                   Enable debug mode"
+    echo "  --debug_pub_key FILE      Set a multiline public key file"
+    echo "  --bump-version            Bump the version in the stemcell with 0.0.<timestamp>"
+    echo
+    echo "Environment Variables:"
+    echo "  stemcell_tgz              Path to the stemcell tarball *REQUIRED*"
+    echo "  AGENT_BINARY              Path to the agent binary to be copied into the stemcell *OPTIONAL*"
+    echo "  AGENT_JSON                Path to the agent JSON configuration file *OPTIONAL*"
+    echo "  BOSH_DEBUG_PUB_KEY        Public key for the BOSH debug user in a single line *OPTIONAL*"
+    echo
+    echo "Example:"
+    echo "  export stemcell_tgz=\"/home/username/workspace/bosh/bosh-linux-stemcell-builder/tmp/bosh-stemcell-0.0.8-google-kvm-ubuntu-noble-go_agent.tgz\""
+    echo "  export AGENT_BINARY=\"/path/to/bosh-agent\""
+    echo "  export AGENT_JSON=\"/path/to/agent.json\""
+    echo "  export BOSH_DEBUG_PUB_KEY=\"ssh-rsa AAA... user@hostname\""
+}
+
+convert_multiline_to_single() {
+    local input_file=$1
+    local single_line_cert
+
+    single_line_cert=$(tr -d '\n' < "$input_file")
+    echo "$single_line_cert"
+}
+
+# Check for help, debug, or debug_pub_key argument
+convert_cert_file=""
+bump_version=false
+for arg in "$@"; do
+    case $arg in
+        --help|-h)
+            show_help
+            exit 0
+            ;;
+        --debug)
+            set -x
+            ;;
+        --debug_pub_key)
+            convert_cert_file="$2"
+            shift 2
+            ;;
+        --bump-version)
+            bump_version=true
+            shift
+            ;;
+        *)
+            shift
+            ;;
+    esac
+done
+
+if [ -n "$convert_cert_file" ]; then
+    BOSH_DEBUG_PUB_KEY=$(convert_multiline_to_single "$convert_cert_file")
+    export BOSH_DEBUG_PUB_KEY
+fi
+
+
+if [ -z ${stemcell_tgz} ]; then
+echo "stemcell_tgz is not set. export stemcell_tgz="/home/username/workspace/bosh/bosh-linux-stemcell-builder/tmp/bosh-stemcell-0.0.8-google-kvm-ubuntu-noble-go_agent.tgz""
```

Comment:
does it make sense that the version is hardcoded here?

Reply:
the version number corresponds with what we use in the readme for building stemcells
https://github.com/cloudfoundry/bosh-linux-stemcell-builder/blob/ubuntu-jammy/README.md?plain=1#L67